src/main/scala/latis/service/hapi/Latis3Interpreter.scala

```diff
@@ -88,6 +96,17 @@ class Latis3Interpreter(catalog: Catalog) extends HapiInterpreter[IO] {
     enc.encode(data)
   }
 
+  override def streamJson(data: T, header: JsonObject): Stream[IO, Byte] = {
+    val dataJson = new JsonEncoder().encode(data)
+    val headerList = header.add("format", Json.fromString("json")).toList
+    val map = ListMap(headerList: _*)
+    val json2Token: Pipe[IO, Json, Token] = fs2.data.json.ast.tokenize
+    val wrapper: Pipe[IO, Token, Token] = fs2.data.json.wrap.asArrayInObject(at = "data", in = map)
+    val renderer: Pipe[IO, Token, String] = fs2.data.json.render.pretty("  ")
+    val toByte: Pipe[IO, String, Byte] = fs2.text.utf8.encode
+    dataJson.through(json2Token).through(wrapper).through(renderer).through(toByte)
```

Comment:
I think this function could be re-written an a for-comprehension and that might be more idiomatic for v3.

Reply:
I wouldn't bother assigning the `Pipe`s to `val`s and just use them in `through` directly.

If you import `fs2.text.utf8` and `fs2.data.json` you can write `json.wrap.asArrayInObject` and `utf8.encode`, which leaves enough context to figure out which pipe you're talking about.

Reply:
> I wouldn't bother assigning the `Pipe`s to `val`s and just use them in `through` directly.
> 
> If you import `fs2.text.utf8` and `fs2.data.json` you can write `json.wrap.asArrayInObject` and `utf8.encode`, which leaves enough context to figure out which pipe you're talking about.

That's fair, I had done it that way for readability and ease of understanding what I was doing when building the series of pipes, but I can see that it would make sense to make the code more compact now.